src/libYARP_OS/src/Node.cpp

```diff
@@ -23,8 +25,38 @@
 using namespace yarp::os;
 using namespace yarp::os::impl;
 
-static ConstString toRosName(const ConstString& str)
-{
+class ROSReport : public PortReport {
+public:
+    std::string sourceURI;
+    std::string targetURI;
+
+    ROSReport() {
+    }
+
+    virtual void report(const PortInfo& info) {
+        NameClient& nic = NameClient::getNameClient();
+
+        std::string sourceTmp = nic.queryName(info.sourceName).toURI();
+        if(sourceTmp!="") {
+            sourceURI = sourceTmp + "/";
```

Comment:
I am not sure whether the `toURI` method should return a "/" at the end by default.